athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchSchemaUtils.java

```diff
@@ -0,0 +1,252 @@
+                metadata.put("scaling_factor", mapping.get("scaling_factor").toString());
+                break;
+            case "float":
+            case "half_float":
+                minorType = Types.MinorType.FLOAT4;
+                break;
+            case "date":
+            case "date_nanos":
+                minorType = Types.MinorType.DATEMILLI;
+                break;
+            case "boolean":
+                minorType = Types.MinorType.BIT;
+                break;
+            default:
+                minorType = Types.MinorType.NULL;
+                break;
+        }
+
+        logger.info("Arrow Type: {}, metadata: {}", minorType.toString(), metadata);
+
+        return new FieldType(true, minorType.getType(), null, metadata);
+    }
+
+    /**
+     * Checks that two mappings are equal irrespective of the ordering of the fields and their children inside
+     * of the Schema objects.
+     * @param mapping1 is a mapping to be compared.
+     * @param mapping2 is a mapping to be compared.
+     * @return true if the lists are equal, false otherwise.
+     */
+    protected final boolean mappingsEqual(Schema mapping1, Schema mapping2)
+    {
+        logger.info("mappingsEqual - Enter:\nMapping1: {}\nMapping2: {}", mapping1, mapping2);
+
+        // Schemas must have the same number of elements.
+        if (mapping1.getFields().size() != mapping2.getFields().size()) {
+            logger.warn("Mappings are different sizes!");
+            return false;
+        }
+
+        // Mappings must have the same fields (irrespective of internal ordering).
+        for (Field field1 : mapping1.getFields()) {
+            Field field2 = mapping2.findField(field1.getName());
+            if (field2 == null || field1.getType() != field2.getType()) {
+                logger.warn("Mapping fields mismatch!");
+                return false;
+            }
+            // The fields' children and metadata maps must also match.
+            logger.info("\nField Name: {}\nField Type: {}", field1.getName(), field1.getType());
+            if (!childrenEqual(field1.getChildren(), field2.getChildren()) ||
+                    !field1.getMetadata().equals(field2.getMetadata())) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Used to assert that children fields inside two mappings are equal.
+     * @param list1 is a list of children fields to be compared.
+     * @param list2 is a list of children fields to be compared.
+     * @return true if the lists are equal, false otherwise.
+     */
+    private final boolean childrenEqual(List<Field> list1, List<Field> list2)
+    {
+        logger.info("childrenEqual - Enter:\nChildren1: {}\nChildren2: {}", list1, list2);
+
+        // Children lists must have the same number of elements.
+        if (list1.size() != list2.size()) {
+            logger.warn("Children lists are different sizes!");
```

Comment:
why is this bad also log more context so we know where it happened (fields?).

Reply:
Added more detailed javadoc, comments, and logs.